oak/common/app_config_serializer.cc

```diff
@@ -41,51 +41,49 @@ int main(int argc, char* argv[]) {
   std::string textproto = absl::GetFlag(FLAGS_textproto);
   if (textproto.empty()) {
     OAK_LOG(FATAL) << "Textproto file is not specified";
+    return 1;
   }
   std::vector<std::string> modules = absl::GetFlag(FLAGS_modules);
   if (modules.empty()) {
     OAK_LOG(FATAL) << "Wasm modules are not specified";
+    return 1;
   }
   std::string output_file = absl::GetFlag(FLAGS_output_file);
   if (output_file.empty()) {
     OAK_LOG(FATAL) << "Output file is not specified";
+    return 1;
   }
 
-  // Parse module names.
+  // Load application configuration.
+  auto config = absl::make_unique<oak::application::ApplicationConfiguration>();
+  std::string textproto_string = oak::utils::read_file(textproto);
+  if (!google::protobuf::TextFormat::MergeFromString(textproto_string, config.get())) {
+    OAK_LOG(FATAL) << "Error parsing proto";
+    return 1;
+  }
+
+  // Parse module names and add Wasm module bytes to the application configuration.
   std::map<std::string, std::string> module_map;
   for (const std::string& module : absl::GetFlag(FLAGS_modules)) {
     std::vector<std::string> module_info = absl::StrSplit(module, ':');
     if (module_info.size() != 2) {
       OAK_LOG(FATAL) << "Incorrect module specification: " << module;
+      return 1;
     }
-    module_map.emplace(module_info.front(), module_info.back());
-  }
-
-  // Load application configuration.
-  auto config = absl::make_unique<oak::application::ApplicationConfiguration>();
-  std::string textproto_string = oak::utils::read_file(textproto);
-  google::protobuf::TextFormat::MergeFromString(textproto_string, config.get());
-
-  // Add Wasm module bytes to the application configuration.
-  for (auto& node_config : *config->mutable_node_configs()) {
-    if (node_config.has_wasm_config()) {
-      std::string module_name = node_config.name();
-      auto it = module_map.find(module_name);
-      if (it != module_map.end()) {
-        std::string module_bytes = oak::utils::read_file(it->second);
-        if (module_bytes.empty()) {
-          OAK_LOG(FATAL) << "Empty Wasm module:" << module_name;
-        }
-        node_config.mutable_wasm_config()->set_module_bytes(module_bytes);
-      } else {
-        OAK_LOG(FATAL) << "Module path for " << module_name << " is not specified";
-      }
+    std::string module_name = module_info[0];
+    std::string module_path = module_info[1];
+    std::string module_bytes = oak::utils::read_file(module_path);
+    if (module_bytes.empty()) {
+      OAK_LOG(FATAL) << "Empty Wasm module:" << module_name;
+      return 1;
     }
+    (*config->mutable_wasm_modules())[module_name] = std::move(module_bytes);
```

Comment:
Can `[module_name]` return `nullptr`?

Reply:
Hopefully it should be obvious if it does, anyways the size of `module_info` was checked above already. Or are you suggesting something else?